Intel Compiler could not use appleframeworks dependency on macOS

**Describe the bug**
When using Intel Compiler (icc) on macOS, `appleframeworks` dependency fails with:

```
Run-time dependency appleframeworks found: NO (tried framework)
```

while icc currently supports `-framework` option:

```
# MrX @ MacBook-Pro in /tmp [22:37:47]
$ icc a.c
Undefined symbols for architecture x86_64:
  "___CFStringMakeConstantString", referenced from:
      _main in icce3u1gw.o
ld: symbol(s) not found for architecture x86_64

# MrX @ MacBook-Pro in /tmp [22:37:49] C:1
$ icc a.c -framework CoreFoundation

# MrX @ MacBook-Pro in /tmp [22:37:58]
$ 
```

**To Reproduce**
meson.build:

```
project('Test', 'c')
dependency('appleframeworks', modules: 'CoreFoundation')
executable('out', 'a.c')
```

Then run `CC=icc meson setup build`.

**Expected behavior**
There should be no issues using frameworks.

Actually, if I change:

https://github.com/mesonbuild/meson/blob/d0325c898ad7f16dc9e22389d365a33097b80af8/mesonbuild/compilers/mixins/clike.py#L1203
to:

```py
        if not (self.id == 'clang' or self.id == "intel"):
```

, meson would correctly found the depencency:

```
Run-time dependency appleframeworks found: YES (CoreFoundation)
```

**system parameters**
* Is this a cross build? - No
* what operating system - macOS 10.15.7
* what Python version are you using - 3.9.5
* what `meson --version` - 0.58.0
* what `ninja --version` if it's a Ninja build - N/A

--- a/mesonbuild/compilers/mixins/clike.py
+++ b/mesonbuild/compilers/mixins/clike.py
@@ -1177,9 +1177,7 @@ class CLikeCompiler(Compiler):
         Finds the framework with the specified name, and returns link args for
         the same or returns None when the framework is not found.
         '''
-        # TODO: maybe this belongs in clang? also, should probably check for macOS?
-        if self.id != 'clang':
-            raise mesonlib.MesonException('Cannot find frameworks with non-clang compiler')
+        # TODO: should probably check for macOS?
         return self._find_framework_impl(name, env, extra_dirs, allow_system)
 
     def get_crt_compile_args(self, crt_val: str, buildtype: str) -> T.List[str]:

